src/ServiceControl/Monitoring/EndpointMonitoringStats.cs

```diff
@@ -0,0 +1,16 @@
+namespace ServiceControl.Monitoring
+{
+    using System.Threading;
+
+    public class EndpointMonitoringStats
+    {
+        private int active;
+        private int failing;
+
+        public int Active => active;
+        public int Failing => failing;
+
+        public void RecordActive() => Interlocked.Increment(ref active);
```

Comment:
GetStats can be called from a web-api call as well (although it will be on a different instance so the interlocked is again redundant). 

Having said that, the status of the endpoints won't update between refreshes and we already have a cached copy of the stats so we could easily just return the cached version and make the update fully sync.